src/zenml/zen_server/rbac/utils.py

```diff
@@ -391,64 +444,128 @@ def get_resource_type_for_model(
-        ServiceConnectorResponse: ResourceType.SERVICE_CONNECTOR,
-        ComponentResponse: ResourceType.STACK_COMPONENT,
-        StackResponse: ResourceType.STACK,
-        PipelineResponse: ResourceType.PIPELINE,
-        CodeRepositoryResponse: ResourceType.CODE_REPOSITORY,
-        SecretResponse: ResourceType.SECRET,
+        ModelRequest: ResourceType.MODEL,
         ModelResponse: ResourceType.MODEL,
+        ModelVersionRequest: ResourceType.MODEL_VERSION,
         ModelVersionResponse: ResourceType.MODEL_VERSION,
-        ArtifactResponse: ResourceType.ARTIFACT,
-        ArtifactVersionResponse: ResourceType.ARTIFACT_VERSION,
-        # WorkspaceResponse: ResourceType.WORKSPACE,
-        # UserResponse: ResourceType.USER,
-        PipelineDeploymentResponse: ResourceType.PIPELINE_DEPLOYMENT,
+        PipelineBuildRequest: ResourceType.PIPELINE_BUILD,
         PipelineBuildResponse: ResourceType.PIPELINE_BUILD,
+        PipelineDeploymentRequest: ResourceType.PIPELINE_DEPLOYMENT,
+        PipelineDeploymentResponse: ResourceType.PIPELINE_DEPLOYMENT,
+        PipelineRequest: ResourceType.PIPELINE,
+        PipelineResponse: ResourceType.PIPELINE,
+        PipelineRunRequest: ResourceType.PIPELINE_RUN,
         PipelineRunResponse: ResourceType.PIPELINE_RUN,
+        RunMetadataRequest: ResourceType.RUN_METADATA,
+        RunTemplateRequest: ResourceType.RUN_TEMPLATE,
         RunTemplateResponse: ResourceType.RUN_TEMPLATE,
+        SecretRequest: ResourceType.SECRET,
+        SecretResponse: ResourceType.SECRET,
+        ServiceAccountRequest: ResourceType.SERVICE_ACCOUNT,
+        ServiceAccountResponse: ResourceType.SERVICE_ACCOUNT,
+        ServiceConnectorRequest: ResourceType.SERVICE_CONNECTOR,
+        ServiceConnectorResponse: ResourceType.SERVICE_CONNECTOR,
+        ServiceRequest: ResourceType.SERVICE,
+        ServiceResponse: ResourceType.SERVICE,
+        StackRequest: ResourceType.STACK,
+        StackResponse: ResourceType.STACK,
+        TagRequest: ResourceType.TAG,
         TagResponse: ResourceType.TAG,
+        TriggerRequest: ResourceType.TRIGGER,
         TriggerResponse: ResourceType.TRIGGER,
+        TriggerExecutionRequest: ResourceType.TRIGGER_EXECUTION,
         TriggerExecutionResponse: ResourceType.TRIGGER_EXECUTION,
-        ServiceAccountResponse: ResourceType.SERVICE_ACCOUNT,
-        ServiceResponse: ResourceType.SERVICE,
+        # WorkspaceResponse: ResourceType.WORKSPACE,
+        # UserResponse: ResourceType.USER,
     }
 
     return mapping.get(type(model))
 
 
-def is_owned_by_authenticated_user(model: AnyResponse) -> bool:
+def is_resource_type_workspace_scoped(resource_type: ResourceType) -> bool:
+    """Check if a resource type is workspace scoped.
+
+    Args:
+        resource_type: The resource type to check.
+
+    Returns:
+        Whether the resource type is workspace scoped.
+    """
+    return resource_type in [
+        ResourceType.STACK,
```

Comment:
This seems wrong still?

Reply:
Good eye! However, this was just a left-over function that was completely unused. I already moved everything to the ResourceType enum class.

It's gone now.